File: crates/auto-lang/test/a2r/020_delegation_params/delegation_params.expected.rs
trait Calculator {
    fn add(&self, a: i32, b: i32) -> i32;
    fn multiply(&self, a: i32, b: i32) -> i32;
}


struct MathEngine {}

impl MathEngine {
    fn add(&self, a: i32, b: i32) -> i32 {
        a + b
    }
    fn multiply(&self, a: i32, b: i32) -> i32 {
        a * b
    }
}

struct Computer {
    engine: MathEngine,
}

impl Calculator for Computer {
    fn add(&self, a: i32, b: i32) -> i32 {
        self.engine.add(a, b)
    }
    fn multiply(&self, a: i32, b: i32) -> i32 {
        self.engine.multiply(a, b)
    }
}

fn main() {
    let comp: Computer = Computer {};
    let result1 = comp.add(5, 3);
    let result2 = comp.multiply(4, 7);
    println!("{}", result1);
    println!("{}", result2);
}

File: crates/auto-lang/test/a2c/13_delegation/003_delegation_params/delegation_params.expected.c
#include "delegation_params.h"


int MathEngine_Add(struct MathEngine *self, int, int) {
    return a + b;
}
int MathEngine_Multiply(struct MathEngine *self, int, int) {
    return a * b;
}

int Computer_add(struct Computer *self, int a, int b) {
    return MathEngine_add(&self->engine, a, b);
}
int Computer_multiply(struct Computer *self, int a, int b) {
    return MathEngine_multiply(&self->engine, a, b);
}

int main(void) {
    struct Computer comp = {};
    unknown result1 = Computer_add(&comp, 5, 3);
    unknown result2 = Computer_multiply(&comp, 4, 7);
    printf("%d\n", result1);
    printf("%d\n", result2);
    return 0;
}
